src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -15,4 +20,18 @@
      * @return {@link User}
      */
     Optional<User> findByUsername(String username);
+
+    @Query("SELECT ur FROM User ur " +
+            "WHERE (:sequence = '' OR LOWER(ur.name) LIKE LOWER(CONCAT('%', :sequence,'%'))) " +
+            "AND (cast(:startDate AS date) IS NULL OR ur.birthDate >= :startDate) " +
+            "AND (cast(:endDate AS date) IS NULL OR ur.birthDate <= :endDate)" )
+
+    /*@Query("SELECT u FROM User u WHERE (:sequence IS NULL OR lower(u.name) like lower(concat('%', :sequence,'%')))"
+            + " and (:startDate IS NULL OR :endDate IS NULL OR u.birthDate BETWEEN :startDate AND :endDate)")*/
+    Page<User> findAllByNameIgnoreCaseContainingAndBirthdateBetween(
```

Comment:
You can remove the "IgnoreCase" to make it shorter 😃 
I think it would be clearer if the order of the parameters matched the order in the name of the method (i.e.: name, startDate and endDate)
This is just a suggestion for neatness :) no need to change it now